Core/src/main/java/com/plotsquared/core/events/PlayerPlotLimitEvent.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * PlotSquared, a land and world management plugin for Minecraft.
+ * Copyright (C) IntellectualSites <https://intellectualsites.com>
+ * Copyright (C) IntellectualSites team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.plotsquared.core.events;
+
+import com.plotsquared.core.player.PlotPlayer;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Called every time after PlotSquared calculated a players plot limit based on their permission.
+ * <p>
+ * May be used to grant a player more plots based on another rank or bought feature.
+ *
+ * @since TODO
+ */
+public class PlayerPlotLimitEvent {
+
+    private final PlotPlayer<?> player;
+
+    private int limit;
+
+    public PlayerPlotLimitEvent(@NonNull final PlotPlayer<?> player, @NonNegative final int limit) {
+        this.player = player;
+        this.limit = limit;
+    }
+
+    /**
+     * Overrides the previously calculated or set plot limit for {@link #player()}.
+     *
+     * @param limit The amount of plots a player may claim. Must be {@code 0} or greater.
+     * @since TODO
+     */
+    public void setLimit(@NonNegative final int limit) {
```

Comment:
The name should probably be `limit` if we're using fluent accessors elsewhere in the class?

Reply:
Totally, yeah - Just used IntelliJ's builder and ignored everything else